uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -2408,6 +2408,11 @@ public function testRecalculateExpiredBalanceChangesWhenAbsenceTypeOfLeaveReques
   public function testCalculateAmountForDateForAContactWithWorkPatternHavingMultipleWeeks() {
     $periodStartDate = new DateTime('2016-01-01');
 
+    AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2016-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2016-12-31')
+    ]);
```

Comment:
Why did you need to add the absence period on these tests? Is it because now the creation of Public Holidays will trigger the recalculation of expired records? If that is the case, what was the performance impact of this during the tests?